### PR TITLE
fix(fa): Add error message when national id is already in use

### PR DESCRIPTION
**Pull request by davidpxn**
# ... 👆

https://app.asana.com/0/1201849781730850/1201814168458673

## What

Check if the national id is already in use and display error message about that.

## Why

So the user know what the error is about.

## Screenshots / Gifs

image

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/financial-aid/web-veita/src/components/NewUserModal/NewUserModal.tsx [hunk 1]
```diff
@@ -93,7 +101,11 @@ const NewUserModal = ({
       setIsVisible={setIsVisible}
       header={'Nýr notandi'}
       hasError={state.hasSubmitError}
-      errorMessage={'Eitthvað fór úrskeiðis, vinsamlega reynið aftur síðar'}
+      errorMessage={
+        state.hasNationalIdError
+          ? 'Mögulega er notandi með þessa kennitölu til nú þegar'
+          : 'Eitthvað fór úrskeiðis, vinsamlega reynið aftur síðar'
+      }
```
Comment (Rikkiabb) [inline apps/financial-aid/web-veita/src/components/NewUserModal/NewUserModal.tsx:108]: Should we rather add the error message to the state and display that?

File: apps/financial-aid/web-veita/src/components/NewUserModal/NewUserModal.tsx [hunk 2]
```diff
@@ -83,7 +85,13 @@ const NewUserModal = ({
         onStaffCreated()
       })
     } catch (e) {
-      setState({ ...state, hasSubmitError: true })
+      setState({
+        ...state,
+        hasSubmitError: true,
+        hasNationalIdError:
+          (e as ApolloError).graphQLErrors[0]?.extensions?.response.status ===
+          400,
```
Comment (Rikkiabb) [inline apps/financial-aid/web-veita/src/components/NewUserModal/NewUserModal.tsx:93]: I know we do it like this in other places, but is there another way of getting the error?

**Review by Rikkiabb — CHANGES_REQUESTED** [review]
One small comment

File: apps/financial-aid/web-veita/src/components/NewUserModal/NewUserModal.tsx (hunk 2 above)
Comment (davidpxn) [reply apps/financial-aid/web-veita/src/components/NewUserModal/NewUserModal.tsx]: This is is only place in the error object that has the status as a number

File: apps/financial-aid/web-veita/src/components/NewUserModal/NewUserModal.tsx [hunk 3]
```diff
@@ -83,7 +85,15 @@ const NewUserModal = ({
         onStaffCreated()
       })
     } catch (e) {
-      setState({ ...state, hasSubmitError: true })
+      setState({
+        ...state,
+        hasSubmitError: true,
```
Comment (Rikkiabb) [inline apps/financial-aid/web-veita/src/components/NewUserModal/NewUserModal.tsx:90]: Do we need `hasSubmitError` ? Can't we just check if errorMessage if undefined? 

That is if we have it by default as undefined

Comment (davidpxn) [reply apps/financial-aid/web-veita/src/components/NewUserModal/NewUserModal.tsx:90]: goood idea😍

**Review by Rikkiabb — APPROVED** [review]
🍸

**Pull request merged**